packages/fxa-auth-server/test/local/payments/subscription-reminders.js

```diff
@@ -0,0 +1,389 @@
+      );
+      sinon.assert.calledOnceWithExactly(
+        mockLog.info,
+        'sendSubscriptionRenewalReminderEmail',
+        {
+          message: 'Sending a renewal reminder email.',
+          subscriptionId: subscription.id,
+          currentPeriodStart: subscription.current_period_start,
+          currentPeriodEnd: subscription.current_period_end,
+          currentDateS: Math.floor(Date.now() / 1000),
+        }
+      );
+      sinon.assert.calledOnceWithExactly(
+        reminder.mailer.sendSubscriptionRenewalReminderEmail,
+        account.emails,
+        account,
+        {
+          acceptLanguage: account.locale,
+        }
+      );
+      sinon.assert.calledOnceWithExactly(
+        reminder.updateSentEmail,
+        subscription.customer.metadata.userid,
+        { subscriptionId: subscription.id }
+      );
+      Date.now = realDateNow;
+    });
+
+    it('returns false if an error is caught when trying to send a reminder email', async () => {
+      const subscription = deepCopy(longSubscription1);
+      subscription.customer = {
+        email: 'abc@123.com',
+        metadata: {
+          userid: 'uid',
+        },
+      };
+      reminder.alreadySentEmail = sandbox.fake.resolves(false);
+      reminder.db.account = sandbox.fake.resolves({});
+      reminder.updateSentEmail = sandbox.fake.resolves({});
+      mockLog.info = sandbox.fake.returns({});
+      mockLog.error = sandbox.fake.returns({});
+      const errMessage = 'Something went wrong.';
+      const throwErr = new Error(errMessage);
+      reminder.mailer.sendSubscriptionRenewalReminderEmail = sandbox.fake.rejects(
+        throwErr
+      );
+      const result = await reminder.sendSubscriptionRenewalReminderEmail(
+        subscription
+      );
+      assert.isFalse(result);
+      sinon.assert.calledOnceWithExactly(
+        reminder.db.account,
+        subscription.customer.metadata.userid
+      );
+      sinon.assert.calledOnceWithExactly(
+        mockLog.error,
+        'sendSubscriptionRenewalReminderEmail',
+        {
+          err: throwErr,
+          subscriptionId: subscription.id,
+        }
+      );
+      sinon.assert.notCalled(reminder.updateSentEmail);
+    });
+  });
+
+  describe('sendReminders', () => {
+    beforeEach(() => {
+      reminder.getEligiblePlans = sandbox.fake.resolves([longPlan1, longPlan2]);
+      reminder.getStartAndEndTimes = sandbox.fake.returns({
+        startTimeS: 1622073600,
+        endTimeS: 1622160000,
+      });
+      async function* genSubscriptionForPlan1() {
+        yield longSubscription1;
+      }
+      async function* genSubscriptionForPlan2() {
+        yield longSubscription2;
+      }
+      mockStripeHelper.findActiveSubscriptionsByPlanId = sandbox.fake(
+        (planId) => {
+          if (planId === longPlan1.plan_id) {
+            return genSubscriptionForPlan1();
+          }
+          return genSubscriptionForPlan2();
+        }
+      );
```

Comment:
There's probably a better way to stub out `mockStripeHelper.findActiveSubscriptionsByPlanId` that asserts which call maps to which arguments. I couldn’t get Sinon’s `stub` API to work.

Reply:
I don't think the mapping matters, or at least I couldn't spot where that would make a difference in the tests.  You could use `stub.onFirstCall` and `stub.onSecondCall`.

Reply:
Thanks; yeah I found a more readable, succinct way to achieve this just now:
```diff
diff --git a/packages/fxa-auth-server/test/local/payments/subscription-reminders.js b/packages/fxa-auth-server/test/local/payments/subscription-reminders.js
index e0d3533ba..4c0c79dc5 100644
--- a/packages/fxa-auth-server/test/local/payments/subscription-reminders.js
+++ b/packages/fxa-auth-server/test/local/payments/subscription-reminders.js
@@ -336,14 +336,12 @@ describe('SubscriptionReminders', () => {
       async function* genSubscriptionForPlan2() {
         yield longSubscription2;
       }
-      mockStripeHelper.findActiveSubscriptionsByPlanId = sandbox.fake(
-        (planId) => {
-          if (planId === longPlan1.plan_id) {
-            return genSubscriptionForPlan1();
-          }
-          return genSubscriptionForPlan2();
-        }
+      const stub = sandbox.stub(
+        mockStripeHelper,
+        'findActiveSubscriptionsByPlanId'
       );
+      stub.onFirstCall().callsFake(genSubscriptionForPlan1);
+      stub.onSecondCall().callsFake(genSubscriptionForPlan2);
     });
     it('returns true if it can process all eligible subscriptions', async () => {
       reminder.sendSubscriptionRenewalReminderEmail = sandbox.fake.resolves({});
(END)
```